src/main/java/org/openrewrite/java/liberty/ChangeMethodInvocation.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.liberty;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.openrewrite.*;
+import org.openrewrite.internal.lang.NonNull;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.*;
+import org.openrewrite.java.tree.JavaType.FullyQualified;
+import org.openrewrite.marker.Markers;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/*
+    Development in this class is not complete
+*/
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class ChangeMethodInvocation extends Recipe {

```

Comment:
I'm a bit puzzled seeing this recipe here; it seems very similar to [the existing](https://docs.openrewrite.org/recipes/java/changemethodname) [ChangeMethodName.java](https://github.com/openrewrite/rewrite/blob/main/rewrite-java/src/main/java/org/openrewrite/java/ChangeMethodName.java)
Is the only change that you had wanted to add a `performStaticCall` option?

Even then we typically see recipes composed from existing recipes, such as for instance here combining:
https://docs.openrewrite.org/recipes/java/changemethodname
https://docs.openrewrite.org/recipes/java/usestaticimport
https://docs.openrewrite.org/recipes/java/changemethodtargettostatic

If you could tell me more about what your intentions were here I might be able to reduce this down to something mostly using existing recipes, such that there's less complexity to maintain here.

Reply:
I was aware of ChangeMethodName, and the other recipes you mentioned. This was one of the first recipes I wrote when I started exploring OpenRewrite. During that investigation, I didn't see an existing recipe that could add new method parameters or change an existing method parameter. The recipe came to be to provide that functionality. I'd be happy to hear that there is a better way to do this.